src/Core/Domain/Shop/ValueObject/ShopConstraint.php

```diff
@@ -52,29 +52,29 @@ class ShopConstraint
     /**
      * Constraint to get configuration for a specific shop
      *
-     * @param int $shopId
+     * @param int|null $shopId
      * @param bool $isStrict
      *
      * @return static
      *
      * @throws ShopException
      */
-    public static function shop(int $shopId, bool $isStrict = false): self
+    public static function shop(?int $shopId, bool $isStrict = false): self
     {
         return new static($shopId, null, $isStrict);
     }
 
     /**
      * Constraint to get configuration for a specific shop group
      *
-     * @param int $shopGroupId
+     * @param int|null $shopGroupId
      * @param bool $isStrict
      *
      * @return static
      *
      * @throws ShopException
      */
-    public static function shopGroup(int $shopGroupId, bool $isStrict = false): self
+    public static function shopGroup(?int $shopGroupId, bool $isStrict = false): self
```

Comment:
what is the reason of this. modification ? it's supposed to be impossible to have a shopGroup or shop context without a related ID

Reply:
I agree this is breaking the interfaces we specifically built to be more strict
Sorry but the problem needs to be fixed at the source, not at the end of the problem I realize it's more complicated 😅

Reply:
Or we need to understand why we have to do that. Legacy code could oblige us to do this dirty fix, but let's be fully aware of the how and why.